### PR TITLE
Add Image Playground Support

### PR DESCRIPTION
**Pull request by jkmassel**
Integrate ImagePlayground in the following screens:
- Gutenberg (React Native)
- Site Icon Picker (My Site)
- Featured Imaged in Post/Page Settings
- Site Media (Add Media)

Outstanding: Gravatar (asked in the respective iOS team channel)

https://github.com/user-attachments/assets/19c65576-fd41-4941-a09e-15b6b1863151

 

To test:

## Regression Notes
1. Potential unintended areas of impact

2. What I did to test those areas of impact (or what existing automated tests I relied on)

3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:

- [ ] I have completed the Regression Notes.
- [ ] I have considered adding unit tests for my changes.
- [ ] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Testing checklist:
- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/wordpress-mobile/WordPress-iOS/milestone/303">25.6 ❄️</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by kean** [comment]
Hey, @crazytonyli , I updated it and added more screens. It's ready for review.

File: WordPress/Classes/ViewRelated/Media/MediaPicker/MediaPickerMenu+ImagePlayground.swift [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        guard #available(iOS 18.1, *) else {
+            return wpAssertionFailure("Not available on this platform. Use `isImagePlaygroundAvailable`.")
+        }
+
+        let controller = _ImagePlaygroundController()
+        controller.delegate = delegate
+
+        let imagePlaygroundVC = ImagePlaygroundViewController()
+        imagePlaygroundVC.delegate = controller
+        objc_setAssociatedObject(imagePlaygroundVC, &MediaPickerMenu.strongDelegateKey, controller, .OBJC_ASSOCIATION_RETAIN_NONATOMIC)
```
Comment (crazytonyli) [inline WordPress/Classes/ViewRelated/Media/MediaPicker/MediaPickerMenu+ImagePlayground.swift:37]: Just to double-check, should `controller` or `imagePlaygroundVC` be stored here?

File: WordPress/Classes/ViewRelated/Media/MediaPicker/MediaPickerMenu+ImagePlayground.swift [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+        presentingViewController.present(imagePlaygroundVC, animated: true)
+    }
+
+    /// ImagePlayground returns heic images that are not supported by many WordPress
+    /// sites. The only exporter that currentyl supports transcoding images is
```
Comment (crazytonyli) [inline WordPress/Classes/ViewRelated/Media/MediaPicker/MediaPickerMenu+ImagePlayground.swift]:
```suggestion
    /// sites. The only exporter that currently supports transcoding images is
```

File: WordPress/Classes/ViewRelated/Media/MediaPicker/MediaPickerMenu+ImagePlayground.swift (hunk 1 above)
Comment (kean) [reply WordPress/Classes/ViewRelated/Media/MediaPicker/MediaPickerMenu+ImagePlayground.swift:37]: `controller` is the value that needs to be retained. `ImagePlaygroundViewController` is kept in memory by UIKit due to being presented on screen.

Comment (crazytonyli) [reply WordPress/Classes/ViewRelated/Media/MediaPicker/MediaPickerMenu+ImagePlayground.swift:37]: So, this line should be ...?
```suggestion
        objc_setAssociatedObject(controller, &MediaPickerMenu.strongDelegateKey, controller, .OBJC_ASSOCIATION_RETAIN_NONATOMIC)
```

Comment (kean) [reply WordPress/Classes/ViewRelated/Media/MediaPicker/MediaPickerMenu+ImagePlayground.swift:37]: The `value` to be retained is a third [parameter](https://developer.apple.com/documentation/objectivec/1418509-objc_setassociatedobject). The test is that the delegate gets called.

Comment (crazytonyli) [reply WordPress/Classes/ViewRelated/Media/MediaPicker/MediaPickerMenu+ImagePlayground.swift:37]: 🙈 Of course... Not sure how I misread that...

**Review by crazytonyli — APPROVED** [review]

File: WordPress/Classes/ViewRelated/Media/MediaPicker/MediaPickerMenu+ImagePlayground.swift (hunk 1 above)
Comment (kean) [reply WordPress/Classes/ViewRelated/Media/MediaPicker/MediaPickerMenu+ImagePlayground.swift:37]: I forget the signature literally every time I use it 😆

**Pull request merged**